src/analysis/convex/function.lean

```diff
@@ -898,3 +898,28 @@ lemma strict_concave_on_iff_div {f : E → β} :
 end has_scalar
 end ordered_add_comm_monoid
 end linear_ordered_field
+
+section
+
+variables [linear_ordered_field 𝕜] [linear_ordered_cancel_add_comm_monoid β] [module 𝕜 β]
+  [ordered_smul 𝕜 β] {x y z : 𝕜} {s : set 𝕜} {f : 𝕜 → β}
+
+lemma convex_on.le_right_of_left_le'' (hf : convex_on 𝕜 s f) (hx : x ∈ s) (hz : z ∈ s)
+  (hxy : x < y) (hyz : y ≤ z) (h : f x ≤ f y) : f y ≤ f z :=
+hyz.eq_or_lt.elim (λ hyz, (congr_arg f hyz).le)
+  (λ hyz, hf.le_right_of_left_le hx hz (Ioo_subset_open_segment ⟨hxy, hyz⟩) h)
+
+lemma convex_on.le_left_of_right_le'' (hf : convex_on 𝕜 s f) (hx : x ∈ s) (hz : z ∈ s)
+  (hxy : x ≤ y) (hyz : y < z) (h : f z ≤ f y) : f y ≤ f x :=
+hxy.eq_or_lt.elim (λ hxy, (congr_arg f hxy).ge)
+  (λ hxy, hf.le_left_of_right_le hx hz (Ioo_subset_open_segment ⟨hxy, hyz⟩) h)
+
+lemma concave_on.le_right_of_left_le'' (hf : concave_on 𝕜 s f) (hx : x ∈ s) (hz : z ∈ s)
```

Comment:
This name and the names of existing lemmas are wrong: actually, it's `right_le_of_le_left`.